editor/src/components/editor/store/dispatch.tsx

```diff
@@ -590,44 +588,50 @@ function editorDispatchInner(
 
     const editorStayedTheSame =
       storedState.nothingChanged &&
-      storedState.editor === result.editor &&
+      storedState.unpatchedEditor === result.unpatchedEditor &&
       storedState.userState === result.userState
 
-    const domMetadataChanged = storedState.editor.domMetadata !== result.editor.domMetadata
-    const spyMetadataChanged = storedState.editor.spyMetadata !== result.editor.spyMetadata
+    const domMetadataChanged =
+      storedState.unpatchedEditor.domMetadata !== result.unpatchedEditor.domMetadata
+    const spyMetadataChanged =
+      storedState.unpatchedEditor.spyMetadata !== result.unpatchedEditor.spyMetadata
     const dragStateLost =
-      storedState.editor.canvas.dragState != null && result.editor.canvas.dragState == null
+      storedState.unpatchedEditor.canvas.dragState != null &&
+      result.unpatchedEditor.canvas.dragState == null
     const metadataChanged = domMetadataChanged || spyMetadataChanged || dragStateLost
     // TODO: Should this condition actually be `&&`?
     // Tested quickly and it broke selection, but I'm mostly certain
     // it should only merge when both have changed.
     if (metadataChanged) {
-      if (result.editor.canvas.dragState != null && 'metadata' in result.editor.canvas.dragState) {
+      if (
+        result.unpatchedEditor.canvas.dragState != null &&
+        'metadata' in result.unpatchedEditor.canvas.dragState
```

Comment:
Is that `result.unpatchedEditor.canvas.dragState.metadata != null` written in an interesting way?

Reply:
> Is that `result.unpatchedEditor.canvas.dragState.metadata != null` written in an interesting way?

yes! the difference is that if metadata is set to `null` or `undefined`, `'metadata' in canvas.dragState` will still return true.

Reply:
I'm not sure. This wasn't part of the change (just the switch from `editor` to `unpatchedEditor`. Will try to find the source of the original code because I'm intrigued now

Reply:
Ah, Balazs has saved me the time there!

Reply:
Exactly, which doesn't seem like we want that.

Reply:
I'm inclined to agree, this seems wrong to me. It was introduced in #2063. Either way, I definitely don't want to address it as part of this PR.

Reply:
In that PR, introduced a new dragState type which has no metadata field. the code here picks which metadata to update: the during-drag-special one (that makes sure the inspector does not re-render) or the regular one. Because in this new dragState there's no metadata field in the dragState, the intention of the code is to pick the regular editorState.jsxMetadata field.

probably a type guard on the dragState type would be more readable/maintainable, but ! we should be removing this code now anyways, because the interactions proposal PR supersedes it